sample-api-server/README.md

```diff
@@ -0,0 +1,218 @@
+`vim $API_HOME/usr/lib/cgi-bin/api.py`  
+`testFailure = True`  
+
+### Production mode / manual scaling
+In a production scaling scenario, the system that does the actual application scaling will need to update the scaling state in the API.  In production mode the cgi script will still set *scalingStatus* from *Successful* to *Pending* as a response to the initial *desiredCapacity* change issued by AWS Auto Scaling.    
+To set production mode:  
+`vim $API_HOME/usr/lib/cgi-bin/api.py`  
+`demoMode = False`  
+
+The state of each *scalableTargetDimensionId* can be set by issuing a PATCH to the API with the appropriate json body payload. 
+ ```
+{"scalingStatus": "Successful", "actualCapacity": 2.0}
+```   
+The current implementation saves state into files in *$API_HOME/var/www/state/* on the local host.  
+For an actual production system, state should be managed in a highly available data store accessible from multiple hosts.  
+This change could be as simple as saving state to an NFS system, or by re-writing the *write_state* and *read_state* functions in *api.py* to talk to a NoSql key-value store.  
+
+### Security considerations for production mode
+*The following steps close security holes that are intentionally open for dev and test.  Follow your own security best practices if you deploy this system.*    
+Disable port 80 on one or more layers of the system  
+* Docker layer  
+Remove *-p 80:80* from the *docker run* statement   
+* Apache layer   
+Disable the Apache VirtualHosts that listen on 80  
+```bash
+docker exec api /usr/sbin/a2dissite 1-api-80.conf  
+docker exec api /usr/sbin/a2dissite 3-api-internal-80.conf
+docker exec api /usr/sbin/apachectl restart  
+```  
+* Block external port 80 traffic in your firewall or AWS VPC Security Group   
+
+Disable access to docroot by editing the relevant virtual host conf files  
+```
+$API_HOME/etc/apache2/sites-available/0-api.conf
+$API_HOME/etc/apache2/sites-available/1-api-80.conf
+$API_HOME/etc/apache2/sites-available/2-api-internal.conf
+$API_HOME/etc/apache2/sites-available/3-api-internal-80.conf
+```
+and un-commenting the following text block   
+```
+#<Directory "/var/www/html/">
+#       Options FollowSymLinks
+#       AllowOverride None
+#       Order Deny,Allow
+#       Deny from All
+#</Directory>
+```   
+Require a host certificate as documented above  
+Apache only supports a single host certificate per VirtualHost so enabling host certificates in 0-api.conf will block all other https client acccess.   
+You can continue to access the api by configuring an alternate dns ServerName in the *2-api-internal.conf* or *3-api-internal-80.conf* conf files.  To use https for internal api access you will need to set up valid SSL certificates.  
+
+To auto-renew the Let's Encrypt certificate, run the following as a scheduled command on your host. 
+`docker exec api /usr/bin/certbot renew`  
```

Comment:
that resolves the auto-renewal issue. How would it get updated on the API Gateway side. Are their command to update that in api gateway

Reply:
This is renewing the LetsEncrypt backend api cert.  Not the API Gateway cert.